### PR TITLE
Adds new check for usage of valid types

### PR DESCRIPTION
**Pull request by MateusJabour**
## What are you adding in this PR?

Adds a new check to make sure that a static block is using a valid block type, one that exists in the directory of blocks.

Read more https://docs.google.com/document/d/1d1EpwfhbgSubLjEqMePZ30ZWJovuCt2mJQslgWSwex0/edit#heading=h.8i0fnctjpka9

## Before you deploy

- [x] This PR includes a new checks or changes the configuration of a check
  - [ ] I included a minor bump `changeset`
  - [ ] It's in the `allChecks` array in `src/checks/index.ts`
  - [ ] I ran `yarn build` and committed the updated configuration files
    
    
    
    - [ ] If applicable, I've updated the `theme-app-extension.yml` config

- [ ] I included a minor bump `changeset`
- [ ] My feature is backward compatible

- [ ] I included a patch bump `changeset`

### CODE REVIEW

**Review by stephanie-shopify — APPROVED** [review]

**Review by arapehl — APPROVED** [review]

File: packages/theme-check-common/src/checks/valid-static-block-type/index.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { doesFileExist } from '../../utils/file-utils';
+
+export const ValidStaticBlockType: LiquidCheckDefinition = {
+  meta: {
+    code: 'ValidStaticBlockType',
+    name: 'Prevent use of type that is not valid for static blocks',
+    docs: {
+      description:
+        'This check is aimed at preventing the use of an invalid type for blocks rendered statically.',
+      url: 'https://shopify.dev/docs/themes/tools/theme-check/checks',
```
Comment (albchu) [inline packages/theme-check-common/src/checks/valid-static-block-type/index.ts:11]: The URL should point at the check's documentation

**Review by albchu — APPROVED** [review]
Documentation link to fix up but otherwise LGTM

**Review by karreiro — COMMENTED** [review]
Thank you, @MateusJabour! I just left one minor comment about the changesets (as they impact the release) :)

File: .changeset/nice-boats-raise.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/theme-check-common': major
```
[the remark this answers is not in the document]
Comment (MateusJabour) [reply .changeset/nice-boats-raise.md:2]: I think it makes sense :)

**Pull request merged**